searchlib/src/tests/memoryindex/field_index/field_index_test.cpp

```diff
@@ -478,14 +478,20 @@ getFeatureStoreMemStats(const FieldIndexCollection &fieldIndexes)
 void
 myCommit(FieldIndexCollection &fieldIndexes, ISequencedTaskExecutor &pushThreads)
 {
+    vespalib::Gate gate;
+    auto gate_callback = std::make_shared<vespalib::GateCallback>(gate);
```

Comment:
Consider using braces to signal lifetime. Easier to match braces than to pair with the reset prior to the await.